### PR TITLE
Parser fixes

### PR DESCRIPTION
**Pull request by jnikula**
Address a couple of recent parser bug reports.

### CODE REVIEW

File: test/function.c [hunk 1]
```diff
@@ -12,3 +12,8 @@ int fooing(void);
  * Empty parameters.
  */
 int fooation();
+
+/**
+ * Array parameters.
+ */
+void foorray(const double array[], int x[5]);
```
Comment (BrunoMSantos) [inline test/function.c:19]: Maybe we can add a couple of more cases for good measure: `[1][2]` and `[1] [2]` (note the space). These should be ok unless clang introduces / keeps spaces between the bracket elements.

**Review by BrunoMSantos — APPROVED** [review]
Nice clean-ups ;)

Had one minor comment about additional test cases, but I think it's pretty solid.

File: test/function.c (hunk 1 above)
Comment (jnikula) [reply test/function.c:19]: Thanks, good idea, added and updated the branch.

**Comment by jnikula** [comment]
And merged, many thanks for the review!

**Pull request merged**